### PR TITLE
Change join program flow

### PR DESCRIPTION
**Pull request by d8660091**
**Description of changes**
- Two steps of join program
- Show welcome banner when join a program

**Type of Change**

- [ ] Bug
- [ ] Styling
- [x] New Feature

**Checklist before requesting review:**

- [ ] Matches design:

  - component sizes, spacing, and styles
  - font size, weight, colour
  - spelling has been double checked

- [ ] New uikit components have Storybook stories
- [ ] Feature is minimally responsive.
- [x] Manual testing of UI feature.
- [ ] Selenium test is completed and passing.

### CODE REVIEW

File: uikit/Tabs/index.tsx [hunk 1]
```diff
@@ -85,7 +95,7 @@ function Tabs({ value, onChange, children: childrenProp }) {
 
   return (
     <TabsContext.Provider value={context}>
-      <Container>{children}</Container>
+      <Container className={className}>{children}</Container>
```
Comment (hlminh2000) [inline uikit/Tabs/index.tsx]: just curious, why should this be removed? seems like a neat feature

File: uikit/notifications/Banner/index.tsx [hunk 2]
```diff
@@ -27,6 +27,7 @@ const Banner = ({
     content={content}
     variant={variant}
     size={size}
+    {...props}
```
Comment (hlminh2000) [inline uikit/notifications/Banner/index.tsx]: if we do this, let's maybe extend the props type to include `& React.ComponentProps<typeof Notification>` so the api is surfaced to consumer :)

File: global/constants/pages.tsx [hunk 3]
```diff
@@ -12,6 +11,8 @@ export const PROGRAM_MANAGE_PATH = `${SUBMISSION_PATH}/program/${PROGRAM_SHORT_N
 export const PROGRAM_DASHBOARD_PATH = `${SUBMISSION_PATH}/program/${PROGRAM_SHORT_NAME_PATH}/dashboard`;
 export const PROGRAM_SAMPLE_REGISTRATION_PATH = `${SUBMISSION_PATH}/program/${PROGRAM_SHORT_NAME_PATH}/sample-registration`;
 export const PROGRAM_CLINICAL_SUBMISSION_PATH = `${SUBMISSION_PATH}/program/${PROGRAM_SHORT_NAME_PATH}/clinical-submission`;
+export const PROGRAM_JOIN_DETAILS_PATH = `${SUBMISSION_PATH}/program/join/details/[inviteId]`;
```
Comment (hlminh2000) [inline global/constants/pages.tsx]: it's a bit verbose, but maybe let's export the `[inviteId]` part as a const and reuse in the pages?

File: uikit/Button/GoogleLogin/index.tsx [hunk 4]
```diff
@@ -15,10 +16,20 @@ const StyledLink = styled('a')`
...
+  id?: string;
+  link: string;
+  className?: string;
+  redirectPath?: string;
+}> = ({ link, id, className, redirectPath }) => {
+  const handleClick = () => {
+    if (!!redirectPath) {
+      localStorage.setItem(LOCAL_STORAGE_REDIRECT_KEY, redirectPath);
+    }
+  };
```
Comment (hlminh2000) [inline uikit/Button/GoogleLogin/index.tsx]: I like the idea of having this logic built-into this button :) only problem is it kind of makes this component app-logic aware, so what do you think of just moving it out of `uikit` and into `components`? 
I think it never belonged here anyway :P

File: components/pages/submission-system/join/details.tsx [hunk 5]
```diff
@@ -54,6 +74,7 @@ export default ({ firstName, lastName, authorizedPrograms = [] }: any) => {
 
       const egoToken = await updateToken(); // Side effect
       if (egoToken) {
+        window.localStorage.setItem('justJoinedProgram', joinProgramInvite.program.shortName);
```
Comment (hlminh2000) [inline components/pages/submission-system/join/details.tsx]: should `"justJoinedProgram"` be in a constant? it's a magic string used by two different file so seems a bit prone to human errors

**Comment by hlminh2000** [comment]
btw TS is complaining :P

File: uikit/Tabs/index.tsx (hunk 1 above)
Comment (hlminh2000) [reply uikit/Tabs/index.tsx]: my bad, it wasn't removed, it was added... sorry brain fart

File: uikit/Button/GoogleLogin/index.tsx (hunk 4 above)
Comment (d8660091) [reply uikit/Button/GoogleLogin/index.tsx]: Totally agree

File: components/pages/submission-system/join/details.tsx (hunk 5 above)
Comment (d8660091) [reply components/pages/submission-system/join/details.tsx]: Good idea

**Pull request merged**